### PR TITLE
Update Changelog

### PR DESCRIPTION
**Pull request by ouziel-slama**

### CODE REVIEW

File: ChangeLog.md [hunk 1]
```diff
@@ -1,4 +1,68 @@
...
+        * Run code scanners:
+            * Pylint
+            * Bandit
+            * CodeQL
+            * License Scanner
+    * Append Only Database:
+        * Add `block_index` field in `balances` table
+        * Remove all UPDATE queries: instead, use the `ledger.insert_update` function which adds a new row with a new `block_index`.
+        * Update all SELECT queries: for a given object identifier we select the line with the `MAX(rowid)`.
+        * Remove completetly `undolog`.
```
Comment (adamkrellenstein) [inline ChangeLog.md:36]: typo

File: ChangeLog.md [hunk 2]
```diff
@@ -1,4 +1,68 @@
...
+    * Rewrite and simplify Dockerfile
+    * Add `simplenode`: Docker compose file to build a minimal node (Bitcoin Core, AddrindexRS and Counterparty Core)
+    * Add Github Workflows. On each push:
+        * Build and install PIP packages then run test suite.
+        * Run code scanners:
+            * Pylint
+            * Bandit
+            * CodeQL
+            * License Scanner
+    * Append Only Database:
```
Comment (adamkrellenstein) [inline ChangeLog.md:32]: Append-Only

**Review by adamkrellenstein — COMMENTED** [review]
This looks great! I think we should also add a list of bugfixes, too, though. Esp. consensus bugs. If for no other reason than to make it clear how important it is that users upgrade.

File: ChangeLog.md (hunk 1 above)
Comment (ouziel-slama) [reply ChangeLog.md]: fixed! thanks!

File: ChangeLog.md [hunk 3]
```diff
@@ -1,4 +1,71 @@
...
+## Counterparty Core Versions ##
+
+* v10.0.0-alpha (2024-03-1?)
+    * Bugfixes
+        * Fix test suite.
+        * Fix software version checking (`check.software_version()`)
+        * Fix database version checking (`check.database_version()`)
+        * DRY and refactor database indexes creation. Fix duplicates index names.
+        * Remove several generic `except`.
+        * Add checkpoints `mainnet` up to 825000 and `tesnet` up to 2540000.
```
Comment (adamkrellenstein) [inline ChangeLog.md:10]: Move to refactoring

File: ChangeLog.md [hunk 4]
```diff
@@ -1,4 +1,71 @@
-## Library Versions ##
+## Counterparty Core Versions ##
+
+* v10.0.0-alpha (2024-03-1?)
+    * Bugfixes
+        * Fix test suite.
```
Comment (adamkrellenstein) [inline ChangeLog.md:5]: Move to refactoring

File: ChangeLog.md [hunk 5]
```diff
@@ -1,4 +1,71 @@
...
+        * Fix software version checking (`check.software_version()`)
+        * Fix database version checking (`check.database_version()`)
+        * DRY and refactor database indexes creation. Fix duplicates index names.
```
Comment (adamkrellenstein) [inline ChangeLog.md:8]: split into bugfix and refactoring

**Review by adamkrellenstein — APPROVED** [review]

**Pull request merged**